aten/src/ATen/core/jit_type.h

```diff
@@ -859,6 +862,14 @@ struct TORCH_API ListType
 
   bool isSubtypeOfExt(const Type& rhs, std::ostream* why_not) const override;
 
+  // global singleton
+  // Given an inner type T and an identifier,
+  // this function wil return the global singleton type pointer
+  // the type List<T>.
+  // The extra "identifier" argument is needed beccause we have multiple container types
+  // that all re-use this function (List<T>, array<T, N>, etc.)
+  static TypePtr get(std::string identifier, TypePtr inner);
```

Comment:
Can we make change the `std::string identifier` to an Enum?

Reply:
We can't make it an enum because of `array<T, N>` - we basically need a different output type for every different container template (and `std::string` gives maybe too much freedom, but I figured it was simple)